scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -100,6 +116,77 @@ public void visit(ImplicitAnonymousFunctionExpressionNode implicitAnonymousFunct
         this.visitSyntaxNode(implicitAnonymousFunctionExpressionNode.expression());
     }
 
+    @Override
+    public void visit(ClassDefinitionNode classDefinitionNode) {
+        checkNonIsolatedPublicClassDefinitions(classDefinitionNode);
+        checkNonIsolatedPublicClassMembers(classDefinitionNode);
+    }
+
+    private void checkNonIsolatedPublicClassDefinitions(ClassDefinitionNode classDefinitionNode) {
```

Comment:
`checkNonIsolatedPublicClassDefinitions` -> `checkNonIsolatedPublicClassDefinition` to show that we are only checking one classdefinition inside the function. Similarly in other place as well